ngmt/modules/td/_pham.py

```diff
@@ -0,0 +1,372 @@
+                            sum_temp = 0  # Reset the temporary sum
+                    except:
+                        f2 = self.index_zero_crossings[i + 1]  # Store the end index of the turn
+                        sum_temp += self.turns_all[i]  # Accumulate the turn angle
+                        turns_no_hesitation.append(sum_temp)  # Store the turn angle without hesitation
+                        flags_start_no_hesitation.append(f1)  # Store the start index of the turn
+                        flags_end_no_hesitation.append(f2)  # Store the end index of the turn
+                        durations_no_hesitation.append((f2 - f1) / sampling_freq_Hz)  # Calculate and store the duration of the turn
+                        z += 1  # Increment the turn index
+                        del f1, f2  # Delete stored indices to avoid conflicts
+                        sum_temp = 0  # Reset the temporary sum
+                else:
+                    sum_temp += self.turns_all[i]  # Accumulate the turn angle if it's smaller than 10 degrees
+            else:  # If there's no hesitation marker at the current index
+                turns_no_hesitation.append(self.turns_all[i])  # Store the turn angle without hesitation
+                flags_start_no_hesitation.append(self.index_zero_crossings[i])  # Store the start index of the turn
+                flags_end_no_hesitation.append(self.index_zero_crossings[i + 1])  # Store the end index of the turn
+                durations_no_hesitation.append((self.index_zero_crossings[i + 1] - self.index_zero_crossings[i]) / sampling_freq_Hz)  # Calculate and store the duration of the turn
+                z += 1  # Increment the turn index
+
+        # Initialize lists to store information about turns >= 90 degrees
+        turns_90 = []  # List to store turn angles >= 90 degrees
+        flags_start_90 = []  # List to store start indices of turns >= 90 degrees
+        flags_end_90 = []  # List to store end indices of turns >= 90 degrees
+
+        # Iterate through each turn without hesitation
+        for k in range(len(turns_no_hesitation)):
+            # Check if the turn angle is greater than or equal to 90 degrees
+            # and if the duration of the turn is between 0.5 and 10 seconds
+            if (abs(turns_no_hesitation[k]) >= 90 and durations_no_hesitation[k] >= 0.5 and durations_no_hesitation [k] < 10):
+                # If conditions are met, store information about the turn
+                turns_90.append(turns_no_hesitation[k])  # Store the turn angle
+                flags_start_90.append(flags_start_no_hesitation[k])  # Store the start index of the turn
+                flags_end_90.append(flags_end_no_hesitation[k])  # Store the end index of the turn
+
+        # Initialize lists to store additional information about >= 90 degree turns
+        duration_90 = []  # List to store durations of >= 90 degree turns
```

Comment:
Perhaps we should separate the detection of a turn, and the calculation of the various spatio-temporal parameters. It would enhance the readability of the code, and is in line with the gait sequence and initial contact detection, where you first detect gait sequences, then initial contacts, and then use that information to obtain stride, swing and stance time.

Reply:
I updated the corresponding part of the calculations and now different spatio-temporal parameters are calculated outside of the detect method. The parameters now could be reached by: 

# Call the spatio_temporal_parameters object for extracting the spatio-temporal parameters
pham.spatio_temporal_parameters()

# Print spatio-temporal parameters for each turn
print(pham.parameters_)